app/src/main/java/com/asfoundation/wallet/nfts/repository/NFTRepository.kt

```diff
@@ -1,19 +1,156 @@
+          .send()
+      var gasLimit = BigInteger.ZERO
+      val gasPrice = web3j.ethGasPrice()
+          .send().gasPrice
+      if (!estimatedGas.hasError()) {
+        gasLimit = estimatedGas.amountUsed
+      }
+      return@fromCallable Pair(gasPrice, gasLimit)
+    }
+        .subscribeOn(Schedulers.io())
+  }
+
+  fun sendNFT(fromAddress: String, toAddress: String, tokenID: BigDecimal, contractAddress: String,
+              schema: String, gasPrice: BigInteger, gasLimit: BigInteger,
+              credentials: Credentials): Single<String> {
+    return Single.fromCallable {
+      signAndSendTransaction(fromAddress, toAddress, tokenID, contractAddress, schema, gasPrice,
+          gasLimit, credentials)
+    }
+        .subscribeOn(Schedulers.io())
+  }
+
+  private fun createTransactionDataNFT721(
+      from: String,
+      to: String,
+      tokenID: BigDecimal,
+  ): ByteArray {
+    val functionName = "transferFrom"
+    val params: List<Type<*>> = listOf(Address(from), Address(to), Uint256(tokenID.toBigInteger()))
+    val returnTypes: List<TypeReference<*>> = listOf(object : TypeReference<Bool?>() {})
+    val function = Function(functionName, params, returnTypes)
+    val encodedFunction = FunctionEncoder.encode(function)
+    return Numeric.hexStringToByteArray(Numeric.cleanHexPrefix(encodedFunction))
+  }
+
+  //TODO fix this function to work with NFT1155
+  private fun createTransactionDataNFT1155(
+      from: String,
+      to: String,
+      tokenID: BigDecimal,
+  ): ByteArray {
+    val functionName = "safeTransferFrom"
+    val params: List<Type<*>> = listOf(Address(from), Address(to), Uint256(tokenID.toBigInteger()),
+        Bytes1(BigInteger.ZERO.toByteArray()))
+    val returnTypes: List<TypeReference<*>> = listOf(object : TypeReference<Bool?>() {})
+    val function = Function(functionName, params, returnTypes)
+    val encodedFunction = FunctionEncoder.encode(function)
+    return Numeric.hexStringToByteArray(Numeric.cleanHexPrefix(encodedFunction))
+  }
+
+  private fun createEstimateGasTransaction(fromAddress: String, toAddress: String,
+                                           tokenID: BigDecimal, contractAddress: String,
+                                           schema: String): Transaction {
+    val data: ByteArray = when (schema) {
+      "ERC721" -> createTransactionDataNFT721(fromAddress, toAddress, tokenID)
+      //"ERC1155" -> createTransactionDataNFT1155(fromAddress, toAddress, tokenID)
+      else -> error("Contract not handled")
+    }
+    val web3j = Web3jFactory.build(HttpService(nftNetwork))
+    val ethGetTransactionCount =
+        web3j.ethGetTransactionCount(fromAddress, DefaultBlockParameterName.LATEST)
+            .send()
+    val nonce = ethGetTransactionCount.transactionCount
+    val gasPrice = web3j.ethGasPrice()
+        .send().gasPrice
+    var gasLimit = BigDecimal(144000).toBigInteger()
+    Log.d("NFT", gasPrice.toString() + "   " + gasLimit.toString())
```

Comment:
An alternative way to build strings in Kotlin is with string templates. 
Here it would look like this: 
"${ gasPrice.toString() }  ${ gasLimit.toString() }"